utils/src/check/clobber-functions.c

```diff
@@ -140,6 +140,53 @@ static struct clobber_function clobber_pb_qconf_wrong_version = {
 	&do_clobber_pb_qconf_wrong_version,
 };
 
+static int pick_random_qslot(void)
+{
+	int j = 0;
+	int filled_slots[SCOUTFS_QUORUM_MAX_SLOTS];
+	uint16_t family;
+	srandom(time(NULL));
+
+	for (int i = 0; i < SCOUTFS_QUORUM_MAX_SLOTS; i++) {
+		family = le16_to_cpu(global_super->qconf.slots[i].addr.v4.family);
+		if (family == SCOUTFS_AF_NONE)
+			continue;
+
+		filled_slots[j] = i;
+		j++;
+	}
+	return filled_slots[(random() & (j-1))];
+}
+
+/*
+ * Clobber a random quorum slot address by setting it to 0
+ */
+static int do_clobber_pb_qslot_bad_addr(char *data)
+{
+	struct scoutfs_quorum_slot *slots = global_super->qconf.slots;
+	int rand_slot = pick_random_qslot();
```

Comment:
This is nice and clean already. But, you could go even further and make the function just return a pointer to the slot directly:
```
struct scoutfs_quorum_slot *slot = pick_random_qslot();
```

The pick_random_slot() then just needs changing so that it returns `global_super->qconf.slots[<random pick>]` or something like that.